### PR TITLE
feat: Use remote OCI image references as rootfs element

### PR DESCRIPTION
**Pull request by nderjung**
### Prerequisite checklist

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [x] Ran `make fmt` on your commit series before opening this PR;
  - [ ] Updated relevant documentation.

### Description of changes

This PR makes it possible to specify a remote OCI image as a rootfs.  This prevents having to use BuildKit in order to retrieve the rootfs, which has become a common operation.

### CODE REVIEW

**Comment by nderjung** [comment]
Pipeline is failing due to a different cause, fixed in https://github.com/unikraft/kraftkit/pull/1631.

File: initrd/ociimage.go [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+	args      []string
+	ref       types.ImageReference
+	env       []string
+	files     []string
+}
+
+// NewFromOCIImage creates a new initrd from a remote container image.
+func NewFromOCIImage(ctx context.Context, path string, opts ...InitrdOption) (Initrd, error) {
+	if !strings.HasPrefix("docker://", path) {
+		path = fmt.Sprintf("docker://%s", path)
```
Comment (craciunoiuc) [inline initrd/ociimage.go]: Can it have other prefixes? I'm scared we could reach cases like `docker://containerd://`

**Review by craciunoiuc — COMMENTED** [review]
only one comment after going through the code

File: initrd/ociimage.go (hunk 1 above)
Comment (nderjung) [reply initrd/ociimage.go]: Yes it can; i'll make it a check for just `://`.

File: initrd/ociimage.go [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+		initrd.files = append(initrd.files, internal)
+
+		log.G(ctx).
+			WithField("file", path).
+			Trace("archiving")
+
+		reader, err := image.OpenPathFromSquash(path)
+		if err != nil {
+			return fmt.Errorf("could not open file: %w", err)
+		}
```
Comment (craciunoiuc) [inline initrd/ociimage.go]: Move this code block inside `:204`  to avoid it failing, that `if` block is the only place where this is used

Comment (craciunoiuc) [reply initrd/ociimage.go]: See PMs

**Comment by craciunoiuc** [comment]
@nderjung  you'll have to rebase again :D

**Review by craciunoiuc — APPROVED** [review]
All good here. Thanks!

Reviewed-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>
Approved-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>

**Pull request merged**